src/frontend/src/optimizer/plan_node/logical_project.rs

```diff
@@ -157,6 +157,17 @@ impl LogicalProject {
                 })
     }
 
+    pub fn try_as_projection(&self) -> Option<Vec<usize>> {
+        self.exprs
+            .iter()
+            .enumerate()
+            .map(|(_i, expr)| match expr {
+                ExprImpl::InputRef(input_ref) => Some(input_ref.index),
+                _ => None,
+            })
+            .collect::<Option<Vec<_>>>()
```

Comment:
In what case will it return None? Shall we report error instead of returning None?